apis/extension/reservation.go

```diff
@@ -67,6 +70,14 @@ type ReservationAffinitySelector struct {
 	ReservationSelectorTerms []corev1.NodeSelectorTerm `json:"reservationSelectorTerms,omitempty"`
 }
 
+type ReservationRestrictedOptions struct {
+	// Resources means that when the Pod intersects with these resources,
+	// it can only allocate the reserved amount at most.
+	// If the Reservation's AllocatePolicy is Restricted, and no resources configured,
+	// by default the resources equal all reserved resources by the Reservation.
+	Resources []corev1.ResourceName `json:"resources,omitempty"`
+}
```

Comment:
Since the default is full restriction, why not use a whitelist to specify the resources that can be relaxed?

Reply:
According to the current definition, the intuition is that by default all resources are unrestricted and the free part of the machine can be used.

Reply:
Controlling restricted lists is easier to maintain.

Reply:
> According to the current definition, the intuition is that by default all resources are unrestricted and the free part of the machine can be used.

The restricted Reservation restricts every reserved resources by default.

Reply:
Well, I got this, but it didn‘t quite fit with intuition.